crml/rewards/src/lib.rs

```diff
@@ -0,0 +1,727 @@
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+
+		fn deposit_event() = default;
+
+		/// Set the per payout inflation rate (`numerator` / `denominator`) (it may be negative)
+		#[weight = (10_000, DispatchClass::Operational)]
+		pub fn set_inflation_rate(origin, numerator: i64, denominator: i64) {
+			ensure_root(origin)?;
+			if denominator.is_zero() {
+				return Err("denominator cannot be zero".into());
+			}
+			InflationRate::put(FixedI128::saturating_from_rational(numerator, denominator));
+		}
+
+		/// Set the development fund take %, capped at 100%.
+		#[weight = (10_000, DispatchClass::Operational)]
+		pub fn set_development_fund_take(origin, new_take_percent: u32) {
+			ensure_root(origin)?;
+			DevelopmentFundTake::put(
+				Perbill::from_percent(new_take_percent) // `from_percent` will saturate at `100`
+			);
+		}
+	}
+}
+
+impl<T: Trait> StakerRewardPayment for Module<T>
+where
+	BalanceOf<T>: FixedPointOperand,
+{
+	type AccountId = T::AccountId;
+	type Balance = BalanceOf<T>;
+	/// Perform a reward payout given a mapping of validators and their nominators stake.
+	/// Accounts IDs are the ones which should receive payment.
+	fn make_reward_payout(
+		validator_commission_stake_map: &[(Self::AccountId, Perbill, Exposure<Self::AccountId, Self::Balance>)],
+	) {
+		// Calculate the accumulated tx fee reward split
+		let fee_payout = TransactionFeePot::<T>::take();
+		// track historic era fee amounts
+		Self::note_fee_payout(fee_payout);
+
+		if fee_payout.is_zero() || validator_commission_stake_map.len().is_zero() {
+			return;
+		}
+
+		let mut total_payout = Self::inflation_rate().saturating_mul_acc_int(fee_payout);
+
+		// Deduct development fund take %
+		let development_fund_payout = Self::development_fund_take() * total_payout;
+		let _ = T::CurrencyToReward::deposit_into_existing(
+			&T::TreasuryModuleId::get().into_account(),
+			development_fund_payout,
+		);
+		total_payout = total_payout.saturating_sub(development_fund_payout);
+
+		// Payout reward to validators and their nominators
+		let total_payout_share = total_payout / BalanceOf::<T>::from(validator_commission_stake_map.len() as u32);
+
+		// implementation note: imbalances have the side affect of updating storage when `drop`ped.
+		// we use `subsume` to absorb all small imbalances (from individual payouts) into one big imbalance (from all payouts).
+		// This ensures only one storage update to total issuance will happen when dropped.
+		let mut total_payout_imbalance = <PositiveImbalanceOf<T>>::zero();
+
+		validator_commission_stake_map
+			.iter()
+			.flat_map(|(validator, validator_commission, stake_map)| {
+				Self::calculate_npos_payouts(&validator, *validator_commission, stake_map, total_payout_share)
+			})
+			.for_each(|(account, payout)| {
+				total_payout_imbalance.maybe_subsume(T::CurrencyToReward::deposit_into_existing(&account, payout).ok());
```

Comment:
I suppose when a fee is paid for an extrinsic, we burn that and that's why we need to create an imbalance for the whole reward payout. Is that true?

Reply:
yes, it will reissue those fee tokens